staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -512,6 +512,62 @@ func PollImmediateUntil(interval time.Duration, condition ConditionFunc, stopCh
 	}
 }
 
+// PollUntilWithContext tries a condition func until it returns true,
```

Comment:
Can you please don't grow this PR further - it's already super huge.

Please split this to a separate PR.

Reply:
move here - https://github.com/kubernetes/kubernetes/pull/101668